packages/reusable-blocks/src/store/test/actions.js

```diff
@@ -1,17 +1,271 @@
+/**
+ * WordPress dependencies
+ */
+import { createRegistry } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import {
+	store as blocksStore,
+	createBlock,
+	registerBlockType,
+	unregisterBlockType,
+} from '@wordpress/blocks';
+
+import { store as coreStore } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+
 /**
  * Internal dependencies
  */
-import { __experimentalSetEditingReusableBlock } from '../actions';
+import { store as reusableBlocksStore } from '../index';
+
+jest.mock( '@wordpress/api-fetch', () => ( {
+	__esModule: true,
+	default: jest.fn(),
+} ) );
+
+function createRegistryWithStores() {
+	// create a registry and register stores
+	const registry = createRegistry();
+
+	registry.register( coreStore );
+	registry.register( blockEditorStore );
+	registry.register( reusableBlocksStore );
+	registry.register( blocksStore );
+
+	// Register entity here instead of mocking API handlers for loadPostTypeEntities()
+	registry.dispatch( coreStore ).addEntities( [
+		{
+			baseURL: '/wp/v2/reusable-blocks',
+			kind: 'postType',
+			name: 'wp_block',
+			label: 'Reusable blocks',
+		},
+	] );
+
+	return registry;
+}
 
 describe( 'Actions', () => {
```

Comment:
So these seem more like integration tests now that the entire store is being tested. I quite like it, it feels more useful to test stores as a standalone module.

Is it worth still keeping separate unit tests for selectors and reducers given they are also tested here?